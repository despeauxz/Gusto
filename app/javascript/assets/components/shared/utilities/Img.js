import React from 'react';
import PropTypes from 'prop-types';


const Img = ({ src, alt, className }) => {
    return (
        <img src={src} alt={alt} className={className} />
    );
};

Img.propTypes = {
    src: PropTypes.string.isRequired,
    alt: PropTypes.string.isRequired,
    className: PropTypes.string
};

Img.defaultProps = {
    className: "" 
};

export default Img;
