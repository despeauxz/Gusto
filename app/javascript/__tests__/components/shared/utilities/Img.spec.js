/* eslint-disable react/jsx-props-no-spreading */
import React from 'react';
import Img from '../../../../assets/components/shared/utilities/Img';


const props = {
    src: "https://placeholder.com/200/300",
    alt: "Placeholder",
    className: ""
};

describe('<Img />', () => {
    it('render the Img component', () => {
        const shallowWrapper = shallow(<Img {...props} />);

        expect(toJson(shallowWrapper)).toMatchSnapshot();
        expect(shallowWrapper.find('Img')).toBeTruthy();
    });
}); 
