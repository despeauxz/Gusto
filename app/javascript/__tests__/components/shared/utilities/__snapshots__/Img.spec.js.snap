// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Img /> render the Img component 1`] = `
<img
  alt="Placeholder"
  className=""
  src="https://placeholder.com/200/300"
/>
`;
